### PR TITLE
Error if you try to run Dynamo compiled function under torch.jit.trace

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111321

Fixes https://github.com/pytorch/pytorch/issues/111319

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by Chillee — APPROVED** [review]

File: torch/_dynamo/eval_frame.py [hunk 1]
```diff
@@ -389,6 +389,15 @@ def _fn(*args, **kwargs):
                 else:
                     return fn(*args, **kwargs)
 
+            if torch.jit.is_tracing():
+                if config.error_on_nested_jit_trace:
```
Comment (bdhirsh) [inline torch/_dynamo/eval_frame.py:393]: If we're using a config to make this toggleable, should we mention the config in the runtime error?

Comment (ezyang) [reply torch/_dynamo/eval_frame.py:393]: Idk I just made it symmetric with the other config lol

**Pull request closed without merging**